internal/interpreter/api.rs

```diff
@@ -2099,6 +2099,58 @@ fn test_multi_components() {
     assert!(result.component("xyz").is_none());
 }
 
+#[cfg(all(test, feature = "highlight"))]
+fn compile(code: &str) -> (ComponentInstance, PathBuf) {
+    i_slint_backend_testing::init_no_event_loop();
+    let mut compiler = Compiler::default();
+    compiler.set_style("fluent".into());
+    let path = PathBuf::from("/tmp/test.slint");
+
+    let compile_result =
+        spin_on::spin_on(compiler.build_from_source(code.to_string(), path.clone()));
+
+    for d in &compile_result.diagnostics {
+        eprintln!("{d}");
+    }
+
+    assert!(!compile_result.has_errors());
+
+    let definition = compile_result.components().next().unwrap();
+    let instance = definition.create().unwrap();
+
+    (instance, path)
+}
+
+#[cfg(feature = "highlight")]
+#[test]
+fn test_element_node_at_source_code_position() {
```

Comment:
since this is testing the highlight feature, i guess it belong in the highlight module, doesn't it?

Reply:
I put it here as this is the entry point I am using, so this has more of an "end to end" test feel to it than just testing some logic away from the interface that gets actually used.